### PR TITLE
changed css visibility utility class to formations

### PR DESCRIPTION
**Pull request by accbjt**
## Description
We've added some visibility classes to Formation, we should be able to replace the Foundation ones in vets-website and design-system and remove the import of the related Foundation sass.

## Testing done
Manual Tests

## Acceptance criteria
- [ ] Sign in name should still show like it did before on mobile and desktop

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by accbjt** [comment]
@jbalboni This was the only thing I could find in this repo and the design-systems repo using the css visibility classes. I did a search for the different variation and this is the only thing I could find.

**Comment by jbalboni** [comment]
@accbjt I see a couple more `hide-for-*` uses (in the signin modal), but it's hard to tell if they're actually dependent on Foundation, since they're also defined in our styles.

Similarly, it looks like we use `show-on-focus`, which is in the foundation visibility file, but we also defined custom styles for it.

For those cases, I would trying removing the foundation visiblity import in core.scss in formation and seeing what the effect is on those instances in vets-website.

**Comment by accbjt** [comment]
@jbalboni I just removed them from the css and replaced the classes with the new formation visibility classes. This should be good to go now.

File: src/platform/user/authentication/components/SignInModal.jsx
Comment (jbalboni) [inline src/platform/user/authentication/components/SignInModal.jsx]: Do you need the `xsmall-screen` one? Also, it seems like it might be `medium-screen:vads-u-display--none` instead of `small-screen` for the second one?

File: src/platform/user/authentication/components/SignInModal.jsx
Comment (jbalboni) [inline src/platform/user/authentication/components/SignInModal.jsx]: There's some weirdness going on with this component. It's mixing together USWDS's grid, Foundation's grid, and our breakpoints and Foundation's breakpoints. As it stands, this change makes this text disappear before it should. It looks like it's supposed to be hidden when the modal shifts to a single column, but this makes it disappear before then. And changing it to `small-screen` doesn't help, because then it doesn't disappear until really narrow widths.

I played around with it a little bit and I think we can fix it by:

1. Leave the classes on this element alone
2. Remove the `columns` and `medium-6` classes on the two divs in this component, along with the `row` class on the div that surrounds them.
3. Remove the right margin on `signin-actions-container`
4. Add a left margin of 2em on `explanation-content`, for medium screen sizes and up.

File: src/platform/site-wide/user-nav/sass/user-nav.scss
Comment (jbalboni) [inline src/platform/site-wide/user-nav/sass/user-nav.scss]: I also want to remove the Foundation imports from Formation. It might be easier to get that in and merged, then make these changes and update package.json in one PR. But it's up to you.

File: src/platform/user/authentication/components/SignInModal.jsx
Comment (accbjt) [reply src/platform/user/authentication/components/SignInModal.jsx]: @jbalboni I made the changes you requested. I added the margin left change for a `$medium-large-screen` because it didn't look right on mobile.

File: src/platform/user/authentication/components/SignInModal.jsx
Comment (jbalboni) [inline src/platform/user/authentication/components/SignInModal.jsx]: Can we remove the columns and medium-6 classes here, too?

**Review by jbalboni — APPROVED** [review]
One more minor comment, but otherwise LGTM

File: src/platform/user/authentication/components/SignInModal.jsx
Comment (accbjt) [reply src/platform/user/authentication/components/SignInModal.jsx]: Sure

**Comment by accbjt** [comment]
@jbalboni I deployed formation 5.0.0 and updated this branch with it. I am now getting errors from the breaking changes made by https://github.com/department-of-veterans-affairs/design-system/commit/bf8ad1ea73ba83fd6fc2a6e33ef3fc9302126932. I think I will need @U-DON to help with fixes this. I see that there are extra props now in the Modal component that are not in the SignInModal component. @U-DON let me know if you are going to fix these breaking changes in another ticket. I don't want to merge anything in until we get all the modals resolved.

**Comment by U-DON** [comment]
The breaking changes will be handled in #9606.

**Comment by jbalboni** [comment]
Yeah, lets wait until @U-DON's PR is merged (I just approved it) and then update this branch.

**Pull request merged**
